docs/rules/func-names.md

```diff
@@ -16,8 +16,8 @@ This rule can enforce or disallow the use of named function expressions.
 
 This rule has a string option:
 
-* `"always"` (default) requires function expressions to have a name
-* `"as-needed"` requires function expressions to have a name, if the name cannot be assigned automatically in an ES6 environment
+* `"always"` (default) requires function expressions and default export functions to have a name
+* `"as-needed"` requires function expressions and default export functions to have a name, if the name cannot be assigned automatically in an ES6 environment
```

Comment:
> Should we add a short note that as-needed requires names for functions in export default, somewhere in the documentation (maybe in the ### as-needed section)?

@mdjermanovic How about this?

Reply:
Hm, I think it still doesn't explain that the `as-needed` option requires a name for default export functions in spite of the fact that the name can be assigned automatically.

Perhaps we could revert this to keep these descriptions short (but slightly inaccurate) and add a note somewhere below?

Maybe something like this at the end of that section (right above `### always`):

```
Please note that `"always"` and `"as-needed"` require function expressions and function declarations in `export default` declarations to have a name.

Reply:
That sounds good.